src/session/session.c

```diff
@@ -2024,3 +2024,79 @@ uint64_t dp_session_unique_id(const struct session *session)
 		return 0;
 	return session->se_id;
 }
+
+/* walk function for query features */
+static int se_feature_query(struct session *s, struct session_feature *sf,
+			    void *data)
+{
+	struct dp_session_info *info = data;
+
+	if (sf->sf_ops && sf->sf_ops->query)
+		sf->sf_ops->query(info, s, sf);
+
+	return 0;
+}
+
+int dp_session_query(struct session *s, enum dp_session_attr query,
+		     struct dp_session_info *info)
+{
+	if (!s)
+		return -1;
+
+	info->se_id = s->se_id;
+
```

Comment:
If `info` was populated with all available information, then `query` wouldn't be necessary which would make it easier to add new information in future.